File: usecases/guest-apiapp-sample/lambda/python/getItem/getItem.py
import json
import os
import boto3
from botocore.exceptions import ClientError
from typing import Any, Dict
from aws_lambda_powertools import Logger, Tracer
from aws_lambda_powertools.utilities.typing import LambdaContext
from aws_lambda_powertools.utilities.data_classes import APIGatewayProxyEvent
from boto3.dynamodb.conditions import Key, Attr

logger = Logger()
tracer = Tracer()

dynamodb = boto3.resource("dynamodb")
table = dynamodb.Table(os.environ["DDB_TABLE"])

@logger.inject_lambda_context(log_event=True)
@tracer.capture_lambda_handler

def lambda_handler(event: APIGatewayProxyEvent, context: LambdaContext) -> Dict[str, Any]:
  logger.info('info')
  logger.debug('debug')

  try:
    response = table.query(
      KeyConditionExpression=Key('title').eq(event['pathParameters']['title']),
    )
    print(response)
  except ClientError as e:
    print(e.response['Error']['Message'])
  else:
    return {
      "statusCode": 200,
      "body" : json.dumps(response["Items"])
    }

File: usecases/blea-guest-serverless-api-sample/lambda/nodejs/getItem.js
'use strict';
const AWSXRay = require('aws-xray-sdk');
const AWS = AWSXRay.captureAWS(require('aws-sdk'));
const ddb = new AWS.DynamoDB();

const tableName = process.env.DDB_TABLE;
module.exports.getItem = (event, context, callback) => {
  var params = {
    KeyConditionExpression: 'title = :title',
    ExpressionAttributeValues: {
      ':title': { S: event.pathParameters.title.toString() },
    },
    TableName: tableName,
  };
  console.log(params);

  ddb.query(params, (error, result) => {
    if (error) {
      console.error(error);
      const response = {
        statusCode: error.statusCode,
        body: error.message,
      };
      callback(null, response);
    } else {
      console.log('Success', result.Items);
      const response = {
        statusCode: 200,
        body: JSON.stringify(result.Items),
      };
      callback(null, response);
    }
  });
};
